community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -450,6 +462,15 @@ private void assertTransactionOpen()
         }
     }
 
+    private void assertOpen()
+    {
+        Optional<Status> terminationReason = getReasonIfTerminated();
+        if ( terminationReason.isPresent() )
+        {
+            throw new TransactionTerminatedException( terminationReason.get() );
+        }
```

Comment:
This is what we test in the old API in KernelStatement. I know for sure that it is not enough to test isOpen. Don't know if we should test both.

Reply:
On top of that it is also a bit confusing since `KernelTransaction::isOpen` returns `true` when the transaction is in state `closing` which is when tx-hooks are run and tx-hooks must be able to run commands so if we cannot add that to check without further refactoring.